cli/proc_state.rs

```diff
@@ -75,6 +76,7 @@ pub struct ProcState(Arc<Inner>);
 pub struct Inner {
   pub dir: DenoDir,
   pub file_fetcher: Arc<FileFetcher>,
+  file_header_overrides: HashMap<ModuleSpecifier, HashMap<String, String>>,
```

Comment:
Do we need to store this here? I believe this could be derived from the CLI options and only need to be stored in fetch_cacher.rs? It seems we could revert most of this file.

Reply:
If we manage to obtain the main module from `flags` and `cli_options`, that should work, good idea.

If we can't obtain the main module from `flags`, are you suggesting we add the main module to `CliOptions`? We'd need to add a parameter to `CliOptions::from_flags(flags, main_module)`, but could remove `file_header_overrides` from `ProcState`.

Sorry for returning the review empty handed. Your feedback is much appreciated!

Reply:
No, it looks like the value of `main_module` could be derived from everything within `CliOptions`. For example, within `CliOptions` when `self.flags.sub_command` is `Eval(_)`, then the main_module can be resolved as `let main_module = resolve_url_or_path("./$deno$eval")?;`. Similarly when it is `Run(run_flags)` then the value can be resolved as:

```
let main_module = if NpmPackageReference::from_str(&run_flags.script).is_ok()
  {
    ModuleSpecifier::parse(&run_flags.script)?
  } else {
    resolve_url_or_path(&run_flags.script)?
  };
```

Reply:
Changed as suggested and reverted most of `proc_state.rs`. Is there a way to avoid all the `map_err(AnyError::from)` I used in `resolve_main_module()`?

Reply:
Yeah, doing this comment should help remove all of them: https://github.com/denoland/deno/pull/17172/files#r1130208510

Reply:
I used `map_err(AnyError::from)` when the return type still was `Result` (Without `Option`) and couldn't figure out how to avoid `map_err`. Without `map_err(AnyError::from)`, the individual `match` arms return different `Results`. Perhaps I need an extra hint here.